storage/replica_command.go

```diff
@@ -756,10 +756,8 @@ func (r *Replica) PushTxn(batch engine.Engine, ms *engine.MVCCStats, args proto.
 		}
 	} else {
 		// Some sanity checks for case where we don't find a transaction record.
-		if args.PusheeTxn.LastHeartbeat != nil {
-			return reply, proto.NewTransactionStatusError(&args.PusheeTxn,
-				"no txn persisted, yet intent has heartbeat")
-		} else if args.PusheeTxn.Status != proto.PENDING {
+		// If `args.PusheeTxn.LastHeartbeat != nil`, assume txn is gc'ed, see #2111.
```

Comment:
This comment doesn't make sense with all references to LastHeartbeat gone from the code. I'd just remove this comment completely.